packages/client/hmi-client/src/components/workflow/tera-workflow.vue

```diff
@@ -465,7 +474,18 @@ function onMenuSelection(operatorType: string, menuNode: WorkflowNode<any>) {
 	newNodePosition.y = menuNode.y;
 
 	if (name && operation && node && drilldown) {
-		addOperatorToWorkflow({ name, operation, node, drilldown })();
+		const newNode: WorkflowNode<any> = addOperatorToWorkflow({ name, operation, node, drilldown })();
+		const inputPort = newNode.inputs.find((input) => input.type.includes(port.type));
```

Comment:
This check will not work complex types, for example: `modelConfigId|datasetId` linking to `modelConfigId|modelId`

Reply:
Also note the menuItem generation is incorrect .... I will see about creating a PR to fix that front.

Reply:
See https://github.com/DARPA-ASKEM/terarium/pull/4502